File: app/src/main/java/com/ocram/qichwadic/presentation/ui/adapters/FavoriteAdapter.java
package com.ocram.qichwadic.presentation.ui.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.core.text.HtmlCompat;
import androidx.recyclerview.widget.RecyclerView;

import com.ocram.qichwadic.R;
import com.ocram.qichwadic.domain.model.Favorite;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class FavoriteAdapter extends RecyclerView.Adapter<FavoriteAdapter.FavoriteViewHolder>{

    private List<Favorite> favorites;
    private FavoriteClickListener listener;

    public FavoriteAdapter(List<Favorite> favorites, FavoriteClickListener listener){
        this.favorites = favorites;
        this.listener = listener;
    }

    @NonNull
    @Override
    public FavoriteViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View v = inflater.inflate(R.layout.item_favorite, parent, false);
        final FavoriteViewHolder viewHolder = new FavoriteViewHolder(v);
        viewHolder.ivShare.setOnClickListener(v1 ->
                listener.onItemShare(favorites.get(viewHolder.getAdapterPosition())));
        viewHolder.ivRemove.setOnClickListener(v2 ->
                listener.onItemRemove(favorites.get(viewHolder.getAdapterPosition())));
        return viewHolder;
    }


    @Override
    public void onBindViewHolder(@NonNull FavoriteViewHolder holder, int position) {
            Favorite currentDefinition = favorites.get(position);
            holder.bindDefinition(currentDefinition);
    }

    public void setFavorites(List<Favorite> favorites) {
        this.favorites = favorites;
    }

    @Override
    public int getItemCount() {
        return favorites.size();
    }

    static class FavoriteViewHolder extends RecyclerView.ViewHolder{

        @BindView(R.id.tvWord) TextView tvWord;
        @BindView(R.id.tvMeaning) TextView tvMeaning;
        @BindView(R.id.ivShare) ImageView ivShare;
        @BindView(R.id.ivRemove) ImageView ivRemove;
        @BindView(R.id.tvDicName) TextView tvDicName;

        FavoriteViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }

        void bindDefinition(final Favorite favorite){
            tvWord.setText(favorite.getWord());
            tvMeaning.setText(HtmlCompat.fromHtml(favorite.getMeaning(), HtmlCompat.FROM_HTML_MODE_LEGACY));
            tvDicName.setText(favorite.getDictionaryName());
        }

    }

    public interface FavoriteClickListener {
        void onItemShare(Favorite definition);
        void onItemRemove(Favorite definition);
    }
}

File: app/src/main/java/com/ocram/qichwadic/features/favorites/ui/FavoriteAdapter.kt
package com.ocram.qichwadic.features.favorites.ui

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.core.text.HtmlCompat
import androidx.recyclerview.widget.RecyclerView

import com.ocram.qichwadic.core.domain.model.DefinitionModel
import com.ocram.qichwadic.databinding.ItemFavoriteBinding

class FavoriteAdapter(private var favorites: List<DefinitionModel>, private val listener: FavoriteClickListener) : RecyclerView.Adapter<FavoriteAdapter.FavoriteViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FavoriteViewHolder {
        val binding = ItemFavoriteBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        val viewHolder = FavoriteViewHolder(binding)
        binding.ivShare.setOnClickListener { favorites[viewHolder.adapterPosition].let { it1 -> listener.onItemShare(it1) } }
        binding.ivRemove.setOnClickListener { favorites[viewHolder.adapterPosition].let { it1 -> listener.onItemRemove(it1) } }
        return viewHolder
    }


    override fun onBindViewHolder(holder: FavoriteViewHolder, position: Int) {
        val currentDefinition = favorites[position]
        holder.bindDefinition(currentDefinition)
    }

    @SuppressLint("NotifyDataSetChanged")
    fun setFavorites(favorites: List<DefinitionModel>) {
        this.favorites = favorites
        notifyDataSetChanged()
    }

    override fun getItemCount(): Int {
        return favorites.size
    }

    class FavoriteViewHolder(val binding: ItemFavoriteBinding) : RecyclerView.ViewHolder(binding.root) {

        fun bindDefinition(favorite: DefinitionModel) {
            binding.tvWord.text = favorite.word
            binding.tvMeaning.text = HtmlCompat.fromHtml(favorite.meaning!!, HtmlCompat.FROM_HTML_MODE_LEGACY)
            binding.tvDicName.text = favorite.dictionaryName
        }

    }

    interface FavoriteClickListener {
        fun onItemShare(definition: DefinitionModel)
        fun onItemRemove(definition: DefinitionModel)
    }
}
